File: src/dataset/go/protein_translation_Exercism_go_ThomasZumsteg/claude/ClaudeSonnet4_protein_translation_v1.go
package protein

var codonMap = map[string]string{
	"AUG": "Methionine",
	"UUU": "Phenylalanine",
	"UUC": "Phenylalanine",
	"UUA": "Leucine",
	"UUG": "Leucine",
	"UCU": "Serine",
	"UCC": "Serine",
	"UCA": "Serine",
	"UCG": "Serine",
	"UAU": "Tyrosine",
	"UAC": "Tyrosine",
	"UGU": "Cysteine",
	"UGC": "Cysteine",
	"UGG": "Tryptophan",
	"UAA": "STOP",
	"UAG": "STOP",
	"UGA": "STOP",
}

func FromCodon(codeon string) string {
	if protein, exists := codonMap[codeon]; exists {
		return protein
	}
	panic("Not a valid codon: " + codeon)
}

func FromRNA(rna string) (codons []string) {
	capacity := len(rna) / 3
	if capacity > 0 {
		codons = make([]string, 0, capacity)
	}
	
	for i := 0; i < len(rna)-2; i += 3 {
		if protein, exists := codonMap[rna[i:i+3]]; exists {
			if protein == "STOP" {
				return
			}
			codons = append(codons, protein)
		} else {
			panic("Not a valid codon: " + rna[i:i+3])
		}
	}
	return
}

File: src/dataset/typescript/protein-translation_Exercism-typescript-FilipeCerejo/claude/ClaudeSonnet4_protein-translation_v1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.translate = translate;
function translate(rna) {
    const codonMap = {
        'AUG': 'Methionine',
        'UUU': 'Phenylalanine',
        'UUC': 'Phenylalanine',
        'UUA': 'Leucine',
        'UUG': 'Leucine',
        'UCU': 'Serine',
        'UCC': 'Serine',
        'UCA': 'Serine',
        'UCG': 'Serine',
        'UAU': 'Tyrosine',
        'UAC': 'Tyrosine',
        'UGU': 'Cysteine',
        'UGC': 'Cysteine',
        'UGG': 'Tryptophan',
        'UAA': null,
        'UAG': null,
        'UGA': null
    };
    const protein = [];
    for (let i = 0; i < rna.length; i += 3) {
        const codon = rna.substr(i, 3);
        const aminoAcid = codonMap[codon];
        if (aminoAcid === null || aminoAcid === undefined) {
            return protein;
        }
        protein.push(aminoAcid);
    }
    return protein;
}
